### PR TITLE
Release v1.2.0

### PR DESCRIPTION
**Pull request by fyrts**

### CODE REVIEW

File: src/ApiAdapter/ApiAdapter.php [hunk 1]
```diff
@@ -63,6 +62,7 @@ public function request(Request $request): ?object
 
         unset($response->Errors);
 
+        $response = (array)$response;
         $response = reset($response);
         return is_object($response) ? $response : null;
```
Comment (DeKerpelLaurent) [inline src/ApiAdapter/ApiAdapter.php:67]: condition always false

File: src/Response/ProjectSummaryResponse.php [hunk 2]
```diff
@@ -11,10 +11,10 @@
...
 class ProjectSummaryResponse extends Response
 {
-    public function __construct($data)
+    public function __construct($data, array $arrays = [])
     {
-        parent::__construct($data, [
+        parent::__construct($data, array_merge([
             'PropertySummaries',
-        ]);
+        ], $arrays));
```
Comment (DeKerpelLaurent) [inline src/Response/ProjectSummaryResponse.php:18]: array_merge 1 array ?

File: src/Response/Response.php [hunk 3]
```diff
@@ -15,11 +15,14 @@ class Response extends ResponseObject
...
      * @param mixed $data
+     *
+     * @return array<static>
      */
     public static function collection($data): array
     {
         $response = [];
         if (!is_array($data)) {
+            $data = (array)$data;
             $data = (reset($data) ?? false) ?: [];
```
Comment (DeKerpelLaurent) [inline src/Response/Response.php]: this will break at some point

File: src/Response/ResponseObject.php [hunk 4]
```diff
@@ -25,7 +26,8 @@ public function __construct(object $data, array $arrays = [])
         foreach ($arrays as $array) {
             if (is_string($array)) {
                 if (isset($data->$array)) {
-                    $value = reset($data->$array);
+                    $value = (array)$data->$array;
```
Comment (DeKerpelLaurent) [inline src/Response/ResponseObject.php]: sure this works ?

**Review by DeKerpelLaurent — CHANGES_REQUESTED** [review]

File: src/Response/ProjectSummaryResponse.php (hunk 2 above)
Comment (fyrts) [reply src/Response/ProjectSummaryResponse.php:18]: array_merge of 2 associative arrays

File: src/ApiAdapter/ApiAdapter.php (hunk 1 above)
Comment (fyrts) [reply src/ApiAdapter/ApiAdapter.php:67]: Depends on the content of $response

**Review by DeKerpelLaurent — APPROVED** [review]

**Pull request merged**